### PR TITLE
Documented NU1004, NU1005, NU1006, NU1007, NU1012 log codes

### PR DESCRIPTION
**Pull request by nkolev92**
Fixes: https://github.com/NuGet/docs.microsoft.com-nuget/issues/1257
Fixes: https://github.com/NuGet/docs.microsoft.com-nuget/issues/2550
Fixes: https://github.com/NuGet/docs.microsoft.com-nuget/issues/2551
Fixes: https://github.com/NuGet/docs.microsoft.com-nuget/issues/2552

One pass of documenting the undocumented log codes.

### CODE REVIEW

File: docs/reference/errors-and-warnings/NU1004.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+- Rerun restore without locked mode to regenerate the lock file.
+- Return the project to it's previous state, removing the additional net5.0 framework.
+
+## Scenario 2
+
+<pre>The package reference Contoso.Library version has changed from [2.0.0, ) to [1.2.3, ).The packages lock file is inconsistent with the project dependencies so restore can't be run
+ in locked mode. Disable the RestoreLockedMode MSBuild property or pass an explicit --force-evaluate option to run restore to update the lock file.</pre>
+
+ ### Issue
```
Comment (zivkan) [inline docs/reference/errors-and-warnings/NU1004.md:36]:
```suggestion
### Issue

```

File: docs/reference/errors-and-warnings/NU1004.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+## Scenario 1
+
+<pre>The project target frameworks are different than the lock file's target frameworks. Lock file target frameworks: net6.0,net5.0. Project target frameworks net6.0.
+The packages lock file is inconsistent with the project dependencies so restore can't be run in locked mode. Disable the RestoreLockedMode MSBuild property or pass an explicit --force-evaluate option to run restore to update the lock file.</pre>
+
+### Issue
+
+When restore is run in locked mode, it ensures that the project dependencies have not changed since the last restore.
+A new framework, `net5.0` has been added to the project.
```
Comment (zivkan) [inline docs/reference/errors-and-warnings/NU1004.md:22]: I don't think there are, but are there any scenarios where running something like `dotnet run -f net5.0` or `dotnet build -f net5.0`, or `dotnet publish -f net5.0` might cause this error to be reported?  Basically, any scenario where telling the dotnet cli to use a specific RID when the project itself supports multiple?

If so, should we call out this scenario specifically?

**Review by zivkan — APPROVED** [review]

**Review by JonDouglas — APPROVED** [review]
LGTM! 🚢 it

File: docs/reference/errors-and-warnings/NU1004.md (hunk 2 above)
Comment (nkolev92) [reply docs/reference/errors-and-warnings/NU1004.md:22]: This shouldn't be happening with net5.0. 

Lock files + rids are challenging though for sure.

**Pull request merged**